internal/provider/models/processors/map_fields.go

```diff
@@ -0,0 +1,136 @@
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"source_field": schema.StringAttribute{
+							Required:    true,
+							Description: "The field to copy data from",
+							Validators: []validator.String{
+								stringvalidator.LengthAtLeast(1),
+							},
+						},
+						"target_field": schema.StringAttribute{
+							Required:    true,
+							Description: "The field to copy data into",
+							Validators: []validator.String{
+								stringvalidator.LengthAtLeast(1),
+							},
+						},
+						"drop_source": schema.BoolAttribute{
+							Optional: true,
+							Computed: true,
+							Default:  booldefault.StaticBool(false),
+							Description: "When enabled, the source field is dropped after the data is copied " +
+								"to the target field. Otherwise, it is preserved.",
+						},
+						"overwrite_target": schema.BoolAttribute{
+							Optional: true,
+							Computed: true,
+							Default:  booldefault.StaticBool(false),
+							Description: "When enabled, any existing data in the target field is overwritten. " +
+								"Otherwise, the target field will be preserved and this mapping will " +
+								"have no effect.",
+						},
+					},
+				},
+			},
+		}),
+	}
+}
+
+func MapFieldsProcessorFromModel(plan *MapFieldsProcessorModel, previousState *MapFieldsProcessorModel) (*Processor, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+	component := &Processor{
+		BaseNode: BaseNode{
+			Type:        "map-fields",
+			Title:       plan.Title.ValueString(),
+			Description: plan.Description.ValueString(),
+			UserConfig:  map[string]any{},
+		},
+	}
+
+	if previousState != nil {
+		component.Id = previousState.Id.ValueString()
+		component.GenerationId = previousState.GenerationId.ValueInt64()
+	}
+
+	component.Inputs = StringListValueToStringSlice(plan.Inputs)
+
+	if !plan.Mappings.IsNull() {
+		mappings := make([]map[string]any, 0)
+		for _, v := range plan.Mappings.Elements() {
+			obj := MapValuesToMapAny(v, &dd)
+			mappings = append(mappings, obj)
+		}
+		component.UserConfig["mappings"] = mappings
+	}
+
+	return component, dd
+}
+
+func MapFieldsProcessorToModel(plan *MapFieldsProcessorModel, component *Processor) {
+	plan.Id = NewStringValue(component.Id)
+	if component.Title != "" {
+		plan.Title = NewStringValue(component.Title)
+	}
+	if component.Description != "" {
+		plan.Description = NewStringValue(component.Description)
+	}
+	plan.GenerationId = NewInt64Value(component.GenerationId)
+	plan.Inputs = SliceToStringListValue(component.Inputs)
+
+	if component.UserConfig["mappings"] != nil {
+		mappings := make([]attr.Value, 0)
+		attrTypes := map[string]attr.Type{
+			"source_field":     StringType{},
+			"target_field":     StringType{},
+			"drop_source":      BoolType{},
+			"overwrite_target": BoolType{},
+		}
```

Comment:
This can be extracted into a package local variable.

Another approach can be to extract the above `map[string]schema.Attribute` into a variable and map it to a `map[string]attr.Type`.